doc/Internals.md

```diff
@@ -460,5 +460,7 @@ So if you need to trigger events in theia:
  - Register events with the emitter.event function
  - Fire events with emitter.fire(event)
 
+# Preferences
 
+You can check out [this page](Preferences.md) to learn more about how preferences work within Theia.
 
```

Comment:
I think we'll have to write more extensive/specific documentation once the plugin system is stable/well defined. For now I think at least having documentation (for both inner workings and how to use it) in the same document is fine. Although we might have to split it up later.